src/orquestra/sdk/_ray/_ray_logs.py

```diff
@@ -125,34 +121,33 @@ def _update_log_filenames(self):
                     )
                 )
 
-    def _read_log_files(self) -> t.Sequence[WFLog]:
-        logs: t.List[WFLog] = []
+    def read_new_logs(self):
+        """
+        Reads log files. Returns all lines produced since last call of this method.
 
+        Updates `self`'s file infos and parsed log objects.
+
+        Yields:
+            Newly parsed log lines that haven't been seen before, as bytes. Each yield
+            relates to one line, as a bytes.
+        """
         self._update_log_filenames()
 
+        log_lines: t.List[bytes] = []
+
         for file_info in self.log_file_infos:
             file_size = os.path.getsize(file_info.filepath)
             if file_size > file_info.size_when_last_opened:
                 file_info.size_when_last_opened = file_size
 
                 with open(file_info.filepath, "rb") as f:
                     f.seek(file_info.file_position)
-                    for next_line in f:
-                        parsed_log = parse_log_line(
-                            next_line, searched_id=self.workflow_or_task_run_id
-                        )
-                        if parsed_log is not None:
-                            logs.append(parsed_log)
-                    file_info.file_position = f.tell()
 
-        return logs
+                    yield from f
+
+                    file_info.file_position = f.tell()
 
-    def get_full_logs(self):
-        parsed_logs = self._read_log_files()
-        formatted = [log.json() for log in parsed_logs]
-        # This is bad. The key should be a task invocation ID. To be fixed in the Jira
-        # ticket: https://zapatacomputing.atlassian.net/browse/ORQSDK-676
-        return {"logs": formatted}
+        return log_lines
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**question:** ‏Can you explain why we return and yield? Whats the example usecase?

Reply:
You've found a leftover after the prev implementation, good catch! I'll update the PR, meanwhile see the motivation below.

The first approach from this PR read all lines from all files, returned them as a one big list, and the lines where being filtered only later on on another function.

The updated approach is a lazy sequence implemented as generator that yields line by line so don't have to keep big lists in memory only to dispose most of them. It's a memory optimization based on gut feeling. This is also similar to the implementation that was there before this PR, so I didn't want to make the performance terribly worse.